Helper/Setup.php

```diff
@@ -134,7 +140,12 @@ public function upgradeData(string $mode, ModuleDataSetupInterface $setup, Modul
                 $moduleData = $this->moduleList->getOne('NS8_Protect');
                 $moduleVersion = $moduleData['setup_version'] ?? '';
                 $storeEmail = $this->scopeConfig->getValue('trans_email/ident_sales/email') ?? '';
-                $storeUrl = rtrim($this->storeManager->getStore()->getBaseUrl(UrlInterface::URL_TYPE_WEB), '/');
+                $storeUrl = rtrim($this->storeManager->getStore()->getBaseUrl(UrlInterface::URL_TYPE_WEB, true), '/');
+                if (substr($storeUrl, 0, strlen(self::MAGENTO_REQUIRED_PROTOCOL)) != self::MAGENTO_REQUIRED_PROTOCOL) {
```

Comment:
I think this could be simplified to:
```
if (strpos($storeUrl, self::MAGENTO_REQUIRED_PROTOCOL) !== 0) {
    ...
}
```
Also if you want to make it case-insensitive, you could use `stripos` instead of `strpos`.

Reply:
@davehensleyns8 Great call-out. I've addressed it in 3ac326b7e48ff4ee64a7a6b10bc67d7da7d90aa2.